pkg/controller/actions/deploy/action_deploy_cache.go

```diff
@@ -94,6 +94,28 @@ func (r *Cache) Has(original *unstructured.Unstructured, modified *unstructured.
 	return exists, nil
 }
 
+func (r *Cache) Delete(original *unstructured.Unstructured, modified *unstructured.Unstructured) error {
+	if original == nil || modified == nil {
+		return nil // nothing to delete
+	}
+
+	key, err := r.computeCacheKey(original, modified)
+	if err != nil {
+		return fmt.Errorf("failed to compute cacheKey for deletion: %w", err)
+	}
+
+	if key == "" {
+		return nil
+	}
+
+	// Check if key exists before deleting to make Delete idempotent
+	_, exists, _ := r.s.GetByKey(key)
```

Comment:
is this actually required ? what would happen if you call Delete for an uncached object ?